src/uct/ib/ud/base/ud_iface.c

```diff
@@ -137,30 +161,48 @@ uct_ud_ep_t *uct_ud_iface_cep_get_ep(uct_ud_iface_t *iface,
     }
 
     conn_match = ucs_conn_match_get_elem(&iface->conn_match_ctx, peer_address,
-                                         conn_sn, queue_type, is_private);
+                                         conn_sn, queue_type, is_rx_only);
     if (conn_match == NULL) {
         return NULL;
     }
 
     ep = ucs_container_of(conn_match, uct_ud_ep_t, conn_match);
     ucs_assert(ep->flags & UCT_UD_EP_FLAG_ON_CEP);
 
-    if (is_private) {
+    if (is_rx_only) {
+        ucs_assertv((ep->flags & UCT_UD_EP_FLAG_TX_RX) == UCT_UD_EP_FLAG_RX,
+                    "ep=%p flags=0x%x", ep, ep->flags);
         ep->flags &= ~UCT_UD_EP_FLAG_ON_CEP;
+    } else {
+        ucs_assertv(ep->flags & UCT_UD_EP_FLAG_TX_RX, "ep=%p flags=0x%x", ep,
+                    ep->flags);
```

Comment:
why needed?

Reply:
if it is found, it should have TX/RX ot both.